symplyphysics/core/experimental/coordinate_systems/__init__.py

```diff
@@ -0,0 +1,199 @@
+"""
+Coordinate systems
+==================
+
+In this module the base class for coordinate systems is defined along with several
+most commonly used systems: Cartesian, cylindrical, and spherical systems.
+
+If you want to add a custom coordinate system, subclass ``CurvilinearCoordinateSystem``
+and define the scalar names and the Lamé coefficients related to it.
+
+**Limitations:**
+
+1. The systems share the same point of origin and are defined in relation to the same
+   axes. The rotation and translation of coordinate systems are not supported.
+
+2. The systems are orthogonal and right-handed.
+
+3. The space is 3-dimensional.
+"""
+
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+
+from sympy import Expr, Symbol, symbols, S, sin
+
+
+class BaseCoordinateSystem(metaclass=ABCMeta):
+    """
+    Abstract base class for all coordinate systems.
+    """
+
+    _display_name: str
+    _base_scalars: tuple[Symbol, Symbol, Symbol]
+    _lame_coefficients: tuple[Expr, Expr, Expr]
+
+    @classmethod
+    @abstractmethod
```

Comment:
My idea was that different types of coordinate systems should be represented by different classes. That is extendable, so the user can roll out their own type of coordinate system if they don't find the one they need presented by us. Therefore, it makes sense to make the base class abstract so that no one could instantiate it since it by definition does not represent any particular type of coordinate systems.

Reply:
This particular method can be not abstract. You can iterate base_scalars and return their string representation as default implementation of this method. Inherited classes can override this behaviour if needed.

Reply:
Hm, okay, I see what you mean.